### ISSUE TITLE
Bot-like features

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/prismarine-web-client#56]
Have some buttons in the ui to:
* Pathfind
* Find block
* Crack world seed
* Auto build schematics

Would be a cool demo for mineflayer and interesting as an user

**Comment by rom1504** [comment]
https://gist.github.com/rom1504/b8530371c7407f790ca63717ee0f8932 + https://github.com/PrismarineJS/prismarine-web-client#some-debugging-examples is a good start

we could have an optional menu (press a key to show it) with some actions like this

**Comment by rom1504** [comment]
put https://github.com/PrismarineJS/prismarine-viewer/blob/master/examples/clickmove.js under a key (point, press g, it goes there)

**Comment by rom1504** [comment]
other idea of interface: in chat use mc command, for example `/goto <player>`

**Comment by rom1504** [comment]
we can have a bot.js to put this kind of thing

**Comment by u9g** [comment]
> other idea of interface: in chat use mc command, for example `/goto <player>`

i disagree with this because then you have chat conflicts

**Comment by u9g** [comment]
> put https://github.com/PrismarineJS/prismarine-viewer/blob/master/examples/clickmove.js under a key (point, press g, it goes there)

I think this should only be done when we finish having remappable keys

**Comment by u9g** [comment]
> https://gist.github.com/rom1504/b8530371c7407f790ca63717ee0f8932 + https://github.com/PrismarineJS/prismarine-web-client#some-debugging-examples is a good start
> 
> we could have an optional menu (press a key to show it) with some actions like this

yeah I think this sounds good

**Comment by AwesomestCode** [comment]
Maybe use a custom prefix like hacked clients? (We’re becoming more like
one every day)

**Comment by rom1504** [comment]
> you have chat conflicts

what kind of conflict?

**Comment by u9g** [comment]
> > you have chat conflicts
> 
> what kind of conflict?

we don't now, but if we add /goto then what if the server had a /goto command too, imo we should stay away from the built in chat box

**Comment by rom1504** [comment]
we can have our own prefix, but imo this is not an issue, it's unlikely to happen (and if it does we can always make it possible to configure the prefix)

**Comment by u9g** [comment]
> we can have our own prefix, but imo this is not an issue, it's unlikely to happen (and if it does we can always make it possible to configure the prefix)

still theres just no reason to use chat commands when minecraft uses them aswell and we can and will have conflicts, just make a menu or something

**Comment by rom1504** [comment]
I guess we could have a different button to open another bar for text commands, but that sounds really non intuitive

not sure what's the point of overengineer this when there are simple solutions

**Comment by rom1504** [comment]
IMO this issue (along with #3 ) is the way to make mineflayer beat baritone, anyone to do it ?
Plan:
* put pweb-client in electron
* add mineflayer menu
* release as installer
* profit

**Comment by rom1504** [comment]
it's about "all contained bot app that works for anyone that doesn't know computers" vs "mineflayer examples/ folder"

**Comment by AwesomestCode** [comment]
Mineflayer still needs some improvements when it comes to physics and
movement. Baritone runs off the Vanilla client so that’s a nonissue for it

**Comment by AwesomestCode** [comment]
Also, why do we need to pack it in electron? IMO running in a browser is an
asset, not a liability.

**Comment by rom1504** [comment]
running in browser alone does not work

**Comment by Heath123** [comment]
This makes it a hacked clientt hough which might make servers ban it and ruin things for people who just want to play the game in a browser

**Comment by rom1504** [comment]
Yeah, that's worth it to make it a mineflayer demo

**Comment by rom1504** [comment]
But anyway that doesn't matter until something works (currently it's not good enough to be ban able)

**Comment by AwesomestCode** [comment]
> This makes it a hacked clientt hough which might make servers ban it and ruin things for people who just want to play the game in a browser

Call it Prismarine Visual Research Platform and send a plugin message to the servers. Allow the servers to deny just PVR.

**Comment by Heath123** [comment]
True, I thought this meant it would be force-enabled by default

**Comment by soshimee** [comment]
I don't think that will do anything, because I think servers are just going to ban them instantly and cause players who just wanted to test to be banned. Also the ones who really want to bypass it would just modify the code locally.
